### PR TITLE
Feature/coin sol

### PR DESCRIPTION
**Pull request by Sotatek-KhanhLe**
----

*Checklist:*

- README.md includes:
  - [ ] getAddress function.
  - [ ] normalTransfer function.
  - [ ] programTransaction function ( solana smart contract ).
  - [ ] splTokenTransaction function.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/coin-sol/src/config/types.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  options?: TransactionOptions;
+};
+
+export type signTxType = {
+  transport: Transport;
+  appPrivateKey: string;
+  appId: string;
+  transaction: TransactionType;
+  confirmCB?: Function;
+  authorizedCB?: Function;
```
Comment (hank121314) [inline packages/coin-sol/src/config/types.ts:42]:
```suggestion
  confirmCB?(): void;
  authorizedCB?(): void;
```

File: packages/coin-sol/src/utils/transactionUtil.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    const value = Number(amount) * (decimals ? Math.pow(10, decimals) : params.LAMPORTS_PER_SOL);
+
+    const hi32 = Math.floor(value / v2e32);
+    const lo32 = value - hi32 * v2e32;
+
+    const programIdIndexSpan = isNormalTransfer ? 4 : 1;
+
+    data.writeUIntLE(programIdToNumber, 0, programIdIndexSpan);
+    data.writeUInt32LE(lo32, programIdIndexSpan);
+    data.writeInt32LE(hi32, programIdIndexSpan + 4);
```
Comment (hank121314) [inline packages/coin-sol/src/utils/transactionUtil.ts:78]: Shouldn't it be `data.writeUInt32LE`? It seems that all `data` layout is `BufferLayout.u64`.  
Please point out if I missing anything.

File: packages/coin-tester/src/utils/componentMaker.tsx [hunk 3]
```diff
@@ -99,6 +99,7 @@ export const TwoInputs = (input: {
               input.setNumberValue(value ? value : 0);
             }
           }}
+          style={{ wordBreak: 'break-words' }}
```
Comment (hank121314) [inline packages/coin-tester/src/utils/componentMaker.tsx:102]: Unrelated change please use `packages/coin-tester/src/components/Inputs.tsx` instead.

File: packages/coin-tester/src/components/coins/index.ts [hunk 4]
```diff
@@ -1,27 +1,29 @@
...
+  // { path: 'etc', Element: ETC },
+  // { path: 'cronos', Element: CRONOS },
+  // { path: 'cro', Element: CRO },
+  // { path: 'matic', Element: MATIC },
+  // { path: 'luna', Element: LUNA },
+  { path: 'SOL', Element: sol },
+  // { path: 'terra', Element: TERRA },
+  // { path: 'xtz', Element: XTZ },
+  // { path: 'bsc', Element: BSC },
 ];
```
Comment (hank121314) [inline packages/coin-tester/src/components/coins/index.ts:29]: Unrelated change, if you only want to build the solana page, please refer to our new `coin-tester` [readme.md](https://github.com/CoolBitX-Technology/coolwallet-sdk/tree/master/packages/coin-tester).

**Review by hank121314 — CHANGES_REQUESTED** [review]

File: packages/coin-sol/src/utils/transactionUtil.ts (hunk 2 above)
Comment (Sotatek-ToanTran) [reply packages/coin-sol/src/utils/transactionUtil.ts:78]: @hank121314  This one we just need to encode each data, the other fields don't need to be used, so we don't have to import BufferLayout

File: packages/coin-tester/src/utils/componentMaker.tsx (hunk 3 above)
Comment (Sotatek-ToanTran) [reply packages/coin-tester/src/utils/componentMaker.tsx:102]: i will update it

File: packages/coin-tester/src/components/coins/index.ts (hunk 4 above)
Comment (Sotatek-ToanTran) [reply packages/coin-tester/src/components/coins/index.ts:29]: Yup, we will update it

File: packages/coin-sol/src/utils/transactionUtil.ts (hunk 2 above)
Comment (hank121314) [reply packages/coin-sol/src/utils/transactionUtil.ts:78]: I mean `hi32` and `lo32` will equal `amount`. 
It will always be an unsigned Integer.
Both `MintToInstructionData ` and `Transfer` are encoded amount as `BufferLayout.u64`.
And `BufferLayout.u64` is encoded in this way https://github.com/pabigot/buffer-layout/blob/326345b0c87d2a98116b22a236ea6623180ffbb1/lib/Layout.js#L750-L759.
So shouldn't it be `data.writeUInt32LE`?
